x-pack/solutions/observability/plugins/observability/public/components/custom_threshold/custom_threshold_rule_expression.tsx

```diff
@@ -348,6 +353,60 @@ export default function Expressions(props: Props) {
     }
   }, [metadata, setRuleParams]);
 
+  const [dashboardList, setDashboardList] = useState<
+    Array<{
+      value: string;
+      label: string;
+    }>
+  >();
+
+  const [selectedDashboards, setSelectedDashboards] = useState<
+    Array<{ label: string; value: string }> | undefined
+  >(
+    ruleParams.dashboards?.map((dashboard) => ({
+      label: dashboard.title,
+      value: dashboard.id,
+    }))
+  );
+
+  const onChange = (selectedOptions: any[]) => {
+    setSelectedDashboards(selectedOptions);
+    setRuleParams(
+      'dashboards',
+      selectedOptions.map((selectedOption) => ({
+        id: selectedOption.value,
+        title: selectedOption.label,
+      }))
+    );
+  };
+
+  const dashboardSavedObjectToMenuItem = (
+    savedObject: SimpleSavedObject<{
+      title: string;
+    }>
+  ) => ({
+    value: savedObject.id,
+    label: savedObject.attributes.title,
+  });
+
+  const loadDashboards = useCallback(
+    async (searchString?: string) => {
+      const { savedObjects: dashboardSO } = await savedObjects.client.find<{ title: string }>({
```

Comment:
@nreese I am using here directly the saved object. Is there a better way to get a list of all dashboards?

Reply:
> I am using here directly the saved object. Is there a better way to get a list of all dashboards?

No, you should not use the savedObjects client to find dashboards. Instead, dashboard plugin start `DashboardStart` exposes [findDashboardsService](https://github.com/elastic/kibana/blob/main/src/platform/plugins/shared/dashboard/public/services/dashboard_content_management_service/types.ts#L106) for this purpose.

Reply:
I removed savedObjects in favor of findDashboardsService in this [commit](https://github.com/elastic/kibana/pull/209668/commits/0ce10271aafa04c09e0026cd31a444cdb45a2b1a)